src/main/java/com/github/dockerjava/api/command/CreateContainerCmd.java

```diff
@@ -197,6 +200,8 @@
     @CheckForNull
     Boolean isTty();
 
+    CreateContainerCmd withAuthConfig(AuthConfig authConfig);
```

Comment:
why it needed for create container?

Reply:
Because docker can pull the image while creating a container, if it is missing on the current host. The auth config is then needed if a private registry is used.